packages/consensus/finality/finality_gadget.go

```diff
@@ -336,10 +342,15 @@ func (s *SimpleFinalityGadget) setMessageGoF(messageMetadata *tangle.MessageMeta
 func (s *SimpleFinalityGadget) setPayloadGoF(messageID tangle.MessageID, gradeOfFinality gof.GradeOfFinality) {
 	s.tangle.Utils.ComputeIfTransaction(messageID, func(transactionID ledgerstate.TransactionID) {
 		s.tangle.LedgerState.TransactionMetadata(transactionID).Consume(func(transactionMetadata *ledgerstate.TransactionMetadata) {
-			// if the transaction is part of a conflicting branch then we need to evaluate based on branch AW
-			if s.tangle.LedgerState.TransactionConflicting(transactionID) {
-				branchID := ledgerstate.NewBranchID(transactionID)
-				gradeOfFinality = s.opts.BranchTransFunc(branchID, s.tangle.ApprovalWeightManager.WeightOfBranch(branchID))
+			// A transaction can't have a higher GoF than its branch, thus we need to evaluate based on min(branchGoF,messageGoF).
+			// This also works for transactions in MasterBranch since it has gof.High and we then use the messageGoF.
+			branchGoF, err := s.tangle.LedgerState.UTXODAG.BranchGradeOfFinality(transactionMetadata.BranchID())
+			if err != nil {
+				// TODO: properly handle error
+				panic(err)
+			}
+			if branchGoF < gradeOfFinality {
+				gradeOfFinality = branchGoF
```

Comment:
Couldn't this cause that a message's GoF overrides a branch GoF which is higher? Imagine a scenario where a branch collected enough weight in other parts of the Tangle through reattachments of the same transaction,

Reply:
Ok, I adjusted this to take reattachments into account via `min(branchGoF,max(messageGoF,transactionGoF))`. This works because the `transactionGoF` should be greater than the current `messageGoF` if there is another attachment with a higher GoF.